pulp_smash/tests/rpm/api_v2/test_remove_unit.py

```diff
@@ -0,0 +1,283 @@
+                ('units after removal', 200),
+        ):
+            with self.subTest(step=step):
+                for response in self.responses[step]:
+                    self.assertEqual(response.status_code, code)
+        for step, code in (
+                ('distribute', 201),
+                ('remove unit', 202),
+        ):
+            with self.subTest(step=step):
+                self.assertEqual(self.responses[step].status_code, code)
+
+    def test_task_error_traceback(self):
+        """Assert each task's "error" and "traceback" fields are null."""
+        for action in {'sync', 'remove unit'}:
+            for i, task in enumerate(self.task_bodies[action]):
+                for key in {'error', 'traceback'}:
+                    with self.subTest((i, key)):
+                        self.assertIsNone(task[key])
+
+    def test_task_progress_report(self):
+        """Assert no task's progress report contains error details."""
+        for i, task in enumerate(self.task_bodies['sync']):
+            with self.subTest(i=i):
+                self.assertEqual(
+                    task['progress_report']['yum_importer']['content']['error_details'],  # noqa pylint:disable=line-too-long
+                    []
+                )
+
+    def test_units_before_removal(self):
+        """Test that units in repositories before removal are the same."""
+        bodies = [re.json() for re in self.responses['units before removal']]
+        # Package category and package group will differ so we count only RPMs
+        self.assertEqual(
+            set(unit['unit_id'] for unit in bodies[0]
+                if unit['unit_type_id'] == 'rpm'),  # This test is fragile
+            set(unit['unit_id'] for unit in bodies[1]
+                if unit['unit_type_id'] == 'rpm'),  # due to hard-coded
+        )  # indices. But the data is complex, and this makes things simpler.
+
+    def test_unit_removed(self):
+        """Test that correct unit from first repository has been removed."""
+        body = self.responses['units after removal'][0].json()
+        units_names = set(unit['metadata']['name'] for unit in body
+                          if unit['unit_type_id'] == 'rpm')
+        self.assertNotIn(self.removed_unit, units_names)
+
+
+class _RemoveMissingTestCase(_BaseTestCase):
+    """Parent class for RemoveMissingTrueTestCase and RemoveMissingFalseTestCase.
+
+    Provides common functionality shared by both child classes.
+    Following steps are executed:
+        1. Create repository foo with feed, sync and publish it.
+        2. Create repository bar with foo as a feed and run sync.
+        3. Get content of both repositories.
+        4. Remove random unit from repository foo and publish foo.
+        5. Sync repository bar.
+        6. Get content of both repositories.
+
+    """
+
+    @classmethod
+    def setUpClass(cls, remove_missing=False):  # noqa pylint:disable=arguments-differ,line-too-long
```

Comment:
I needed to disable `arguments-differ` warning and by adding that line got 81 chars long, I didn't know though that it disables `line-too-long` warning for whole method. Renaming `remove_missing` to e.g. `rm_missing` would do the trick.